src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -324,5 +327,24 @@ public Task<bool> DelayUntilHostReady(int timeoutSeconds = ScriptConstants.HostT
 
             return hostReady;
         }
+
+        internal static async Task<bool> DelayUntilHostReady(WebHostResolver resolver)
```

Comment:
Would be really nice if we could avoid exposing more static methods like this. This is a challenge to some of the DI work coming down the pipe.

Reply:
I don't understand - what's the problem with static helper methods? We have lots of them and they're standard programming practice. E.g. extension methods, etc. I could put this as an extension method off of WebHostResolver but that is another static method, and it doesn't really belong there. I factored this code out to keep the other method simple.

Reply:
The problem we have everywhere here is shared static state, which is always a source of issues. :)